app/controllers/APIController.scala

```diff
@@ -35,4 +35,22 @@ class APIController @Inject() (
         Ok(databaseTables)
       }
   }
+
+//  def createProduct() = Action.async { implicit request: Request[AnyContent] =>
+//    Ok("new item added")
+//  }
+
+  def getProduct(id: String) = Action.async {
+    implicit request: Request[AnyContent] =>
+      bakeryDB.getProductById(id).map {
+        case ""                => NotFound("No Product Found")
+        case myProduct @ found => Ok(myProduct.toString)
+      }
```

Comment:
The preferred way in which we will treat emptyness in scala (as opposed to traditional java; not the modern one) will be with `Option`. To take advantage of them, you have to define your function to return a `Future[Option[String]]`